### PR TITLE
Feat/Goblin Mint/Bank Updates

### PR DESCRIPTION
**Pull request by spencerdezartsmith**
# Description
This PR aims to add a narrative to the cooldown period that comes with minting a [rare item](https://docs.sunflower-land.com/crafting-guide/farming-and-gathering#crafting-limits). The story being that goblins are expert craftsman and need 7 days to complete the building of your rare item.

Note: I am working with assets that have not be finalised yet.

This PR also adds some UI updates for the bank modals. You can now move between withdrawing tokens and items easily. These modals need more work in the future but this is a good first step. 
Screen Shot 2022-05-07 at 11 34 48 am

Screen Shot 2022-05-07 at 11 34 01 am

Screen Shot 2022-05-07 at 11 35 48 am

Screen Shot 2022-05-07 at 11 38 45 am

I have also added an `error` color to tailwind config.a

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- Go to Goblin Land
- Mint and item
- Confirm the cooldown period is visible. Does the story make sense?
- Check out the bank modals

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [x] Screenshot if it includes any UI changes
- [ ] I have read the contributing guidelines and agree to the T&Cs
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by brian-man** [comment]
@spencerdezartsmith Kind of breaks "you own your stuff" if you can't immediately withdraw after crafting. I get the intent, but it feels very anti-blockchain.

**Comment by adamhannigan** [comment]
@spencerdezartsmith Love the goblin!

Similar to comments by @brian-man, we should shift away from the Goblin's "holding" your assets and reword it as the Goblins are "building" your assets.

Similar to most games, to craft/build an item takes time. For the player this will feel less like a negative and more just like gameplay

File: src/features/goblins/bank/components/WithdrawItems.tsx [hunk 1]
```diff
@@ -97,65 +113,102 @@ export const WithdrawItems: React.FC<Props> = ({ onWithdraw }) => {
...
-        {inventoryItems.length < 4 &&
-          new Array(4 - inventoryItems.length)
-            .fill(null)
-            .map((_, index) => <Box disabled key={index} />)}
-      </div>
-
-      <div className="mt-2">
-        <span className="text-shadow text-base">Selected</span>
+      <div className="mt-3">
+        <div className="flex items-center border-2 rounded-md border-black p-2 bg-green-background mb-3">
```
Comment (adamhannigan) [inline src/features/goblins/bank/components/WithdrawItems.tsx:117]: Screen Shot 2022-05-07 at 8 53 35 am

I would suggest omitting the green text part and opting for using the timer icon on top of the icon you previously had. It should be clear to the players that an item takes X time to be crafted.

On my laptop, the modal height was very jarring and took up most of the page. The 3 warnings on this page was a bit overwhelming as well.

**Review by adamhannigan — COMMENTED** [review]
Nice flows mate 💪 

Code looks good, I've just added some suggestions around the wording we are using. We want to move away from the concept of items being "held" as it has a negative connotation to it. The 'crafting' time should be a natural gameplay element that does not need much explanation to the user.

File: src/features/goblins/bank/components/WithdrawItems.tsx (hunk 1 above)
Comment (spencerdezartsmith) [reply src/features/goblins/bank/components/WithdrawItems.tsx:117]: Yep cool. I have changed the narrative. I would argue that the green sign stays. I don't think it is immediately clear as to why you can't withdraw some items. Especially the ones that are "in use". Its not immediately clear that a scarecrow is in use just because crops are planted. It would also be easy to miss that something is still in the 7 day cooldown period. People may not pay attention to that and then you come here and all the sudden they're disabled. It doesn't give a lot of feedback.

I have removed the second red warning. See what you think. If you still aren't convinced I will remove the green.

**Comment by spencerdezartsmith** [comment]
> @spencerdezartsmith Kind of breaks "you own your stuff" if you can't immediately withdraw after crafting. I get the intent, but it feels very anti-blockchain.
> 
> The withdraw holding period doesn't affect whales/profiteers as they already have to wait a week because of the crafting cooldown.
> 
> This would only affect the "smaller" players who craft a single item with intent to sell it. Adding a holding period of a week for them, which might mean the market changes drastically between those two points in time. As an example, if a small player wants to craft a Woody to address current demand that demand may vanish in a week.
> 
> TLDR: I'm not seeing the benefit to the withdraw period at all, but I do see harm.

Hey there... this was already a proposal that was accepted. I am just implementing it 😄 . Someone has linked the proposal to this PR.

File: src/features/goblins/Rare.tsx [hunk 2]
```diff
@@ -141,33 +149,44 @@ export const Rare: React.FC<Props> = ({ onClose, items, canCraft = true }) => {
...
+            </p>
+            <p className="text-[10px]">
+              {`It will be finished in ${secondsToString(
+                selected.cooldownSeconds || 0
+              )}.`}
+            </p>
+          </div>
+          <div className="my-3 text-center">
+            <a
+              href={`https://docs.sunflower-land.com/crafting-guide/farming-and-gathering#crafting-limits`}
```
Comment (adamhannigan) [inline src/features/goblins/Rare.tsx:182]: You've been added to gitbook for when you need to update the docs 😄

**Review by adamhannigan — APPROVED** [review]
LGTM. The green message looks good now we don't have the 2 red messages. I would also consider using the grey background for this message since the green often represents a success message (when the message is not exactly a bonus 😅 )

I'm having some issues minting but will get this branch in so

**Comment by brian-man** [comment]
> Hey there... this was already a proposal that was accepted. I am just implementing it 😄 . Someone has linked the proposal to this PR.

Just sharing an opinion.  I think the 

Critique was directed at the product direction, not at the messenger. :)

@spencerdezartsmith
